File: SafeTrace_App/SafeTrace/ios/MPCModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MPCModule, NSObject)

RCT_EXTERN_METHOD(splitShares:(NSString *) sequence
                  resolver:(RCTPromiseResolveBlock) resolve
                  rejecter:(RCTPromiseRejectBlock) reject)

RCT_EXTERN_METHOD(joinShares:(NSArray *) shares
                  resolver:(RCTPromiseResolveBlock) resolve
                  rejecter:(RCTPromiseRejectBlock) reject)

@end

File: SafeTrace_App/SafeTrace/ios/MPCModule.swift
import Foundation

@objc(MPCModule)
class MPCModule : NSObject {
  
  @objc(splitShares:resolver:rejecter:)
  func splitShares(_ sequence: String, resolver resolve: RCTPromiseResolveBlock, rejecter reject:RCTPromiseRejectBlock) {
    resolve(true);
  }
  
  @objc(joinShares:resolver:rejecter:)
   func joinShares(_ shares: NSArray, resolver resolve: RCTPromiseResolveBlock, rejecter reject:RCTPromiseRejectBlock) {
     resolve(true);
   }
  
}
